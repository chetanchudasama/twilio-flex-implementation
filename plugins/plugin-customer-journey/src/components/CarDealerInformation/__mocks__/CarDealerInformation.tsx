import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
