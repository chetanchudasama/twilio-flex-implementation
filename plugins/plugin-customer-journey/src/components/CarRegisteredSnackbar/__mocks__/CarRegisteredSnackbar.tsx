import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
