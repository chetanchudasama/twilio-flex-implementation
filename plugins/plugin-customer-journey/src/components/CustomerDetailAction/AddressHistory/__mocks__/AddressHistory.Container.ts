import AddressHistory from "./AddressHistory";

export default AddressHistory;
