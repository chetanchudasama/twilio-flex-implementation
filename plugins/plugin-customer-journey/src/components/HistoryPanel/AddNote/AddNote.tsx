import {
  Button,
  Checkbox,
  FormControlLabel,
  FormGroup,
  Grid,
  Icon,
  TextField,
} from "@material-ui/core";
import CheckBoxIcon from "@material-ui/icons/CheckBox";
import CheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import React, { useEffect, useState } from "react";
import { AddNoteDetailModel } from "../../../models/AddNoteDetailModel";
import { AddNoteStyles } from "./AddNote.Styles";

export interface AddNoteProps {
  isNoteAdded: boolean;
  onSaveNoteDetail: (addNoteModel: AddNoteDetailModel) => void;
}

const AddNote: React.FC<AddNoteProps> = ({ isNoteAdded, onSaveNoteDetail }) => {
  const [addNoteDetail, setAddNoteDetail] = useState<AddNoteDetailModel>(
    new AddNoteDetailModel()
  );

  useEffect(() => {
    setAddNoteDetail(new AddNoteDetailModel());
  }, [isNoteAdded]);

  const handleAddNoteDetailChange = (event: any, propertyName: string) => {
    setAddNoteDetail({
      ...addNoteDetail,
      [propertyName]:
        propertyName === "content" ? event.target.value : event.target.checked,
    });
  };

  return (
    <AddNoteStyles>
      <Grid container>
        <Grid className="add-note-container" item xs={10}>
          <Grid item xs={12}>
            <TextField
              className="add-note-text-field"
              onChange={(event) => handleAddNoteDetailChange(event, "content")}
              multiline
              rows={3}
              InputProps={{
                placeholder: "Add note",
              }}
              variant="standard"
              value={addNoteDetail.content}
            />
          </Grid>
          <Grid item xs={12}>
            <FormGroup row>
              <FormControlLabel
                className="add-note-form-control"
                control={
                  <Checkbox
                    icon={
                      <CheckBoxOutlineBlankIcon
                        fontSize="small"
                        className="checkbox"
                      />
                    }
                    checkedIcon={
                      <CheckBoxIcon fontSize="small" className="checkbox" />
                    }
                    checked={addNoteDetail.isImportant}
                    onChange={(event) =>
                      handleAddNoteDetailChange(event, "isImportant")
                    }
                    name="Important"
                  />
                }
                label="Important"
              />
              <FormControlLabel
                className="add-note-form-control"
                control={
                  <Checkbox
                    icon={
                      <CheckBoxOutlineBlankIcon
                        fontSize="small"
                        className="checkbox"
                      />
                    }
                    checkedIcon={
                      <CheckBoxIcon fontSize="small" className="checkbox" />
                    }
                    checked={addNoteDetail.sendToBroker}
                    onChange={(event) =>
                      handleAddNoteDetailChange(event, "sendToBroker")
                    }
                    name="Share with broker"
                  />
                }
                label="Share With Broker"
              />
            </FormGroup>
          </Grid>
        </Grid>
        <Grid item xs={2}>
          <Button
            variant="contained"
            color="primary"
            className={`send-btn ${
              !addNoteDetail.content.trim() ? " disable-btn" : ""
            }`}
            onClick={() => onSaveNoteDetail(addNoteDetail)}
            disabled={!addNoteDetail.content.trim()}
          >
            <Icon className="send-icon">send</Icon>
          </Button>
        </Grid>
      </Grid>
    </AddNoteStyles>
  );
};

export default AddNote;
