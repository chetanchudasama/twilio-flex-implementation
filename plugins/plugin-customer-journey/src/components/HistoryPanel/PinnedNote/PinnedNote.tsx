import { Grid, Icon } from "@material-ui/core";
import React from "react";
import { HistoryPanelDetailModel } from "../../../models/HistoryPanelDetailModel";
import { PinnedNoteStyles } from "./PinnedNote.Styles";

export interface PinnedNoteProps {
  pinnedNoteList: HistoryPanelDetailModel[];
  handleUnpinNote: (id: string) => void;
  containerHeight: number;
}

const PinnedNote: React.FC<PinnedNoteProps> = ({
  pinnedNoteList,
  handleUnpinNote,
  containerHeight,
}) => {
  return (
    <PinnedNoteStyles>
      <div
        className="pinned-chat-container"
        style={{ height: `${containerHeight}px` }}
      >
        {pinnedNoteList.map((message) => (
          <Grid container className="pinned-content" key={message.id}>
            <Grid item xs={1}>
              <Icon className="customer-icon-position" fontSize="small">
                check_circle
              </Icon>
            </Grid>
            <Grid item xs={10} className="message-content">
              {message.content}
            </Grid>
            <Grid item xs={1}>
              <Icon
                className="icon-close"
                fontSize="small"
                onClick={() => handleUnpinNote(message.id)}
              >
                close
              </Icon>
            </Grid>
          </Grid>
        ))}
      </div>
    </PinnedNoteStyles>
  );
};

export default PinnedNote;
