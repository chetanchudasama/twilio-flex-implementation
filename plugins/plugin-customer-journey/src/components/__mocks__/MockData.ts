import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
