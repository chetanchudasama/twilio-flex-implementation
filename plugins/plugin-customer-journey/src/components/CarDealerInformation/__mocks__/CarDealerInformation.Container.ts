import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
