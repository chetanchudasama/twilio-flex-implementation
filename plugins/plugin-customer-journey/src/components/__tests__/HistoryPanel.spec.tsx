import React from "react";
import { Manager } from "@twilio/flex-ui";
import { create } from "react-test-renderer";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core";
import { Shared, ImagePreview, DocumentDownload } from "@common/components";
import DirectionsCarIcon from "@material-ui/icons/DirectionsCar";
import PhoneInTalkIcon from "@material-ui/icons/PhoneInTalk";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { HistoryPanelMemberType } from "../../common/enum";
import PinnedNote from "../HistoryPanel/PinnedNote/PinnedNote";
import HistoryPanel, { HistoryPanelProps } from "../HistoryPanel/HistoryPanel";
import {
  StateToProps,
  DispatchToProps,
} from "../HistoryPanel/HistoryPanel.Container";
import { AddNoteDetailModel } from "../../models/AddNoteDetailModel";
import AddNote from "../HistoryPanel/AddNote/AddNote";
import { flexManagerMock } from "../__mocks__/MockData";

type Props = StateToProps & DispatchToProps & HistoryPanelProps;

const props: Props = {
  historyPanelList: [
    {
      id: "1",
      actionType: {
        id: 1,
        name: "SMS",
      },
      actionedBy: 1,
      content: "Hello",
      isImportant: false,
      timestamp: new Date("06-02-2021 13:24"),
      url: "",
    },
    {
      id: "2",
      actionType: {
        id: 1,
        name: "call",
      },
      actionedBy: 2,
      content: "Hey, thanks for reaching out. How may I help you today?",
      isImportant: false,
      timestamp: new Date("06-02-2021 13:24"),
      url: "",
    },
    {
      id: "3",
      actionType: {
        id: 1,
        name: "recordOpened",
      },
      actionedBy: 2,
      content: "Record opened, Dan Graham",
      isImportant: false,
      timestamp: new Date("06-02-2021 13:24"),
      url: "",
    },
    {
      id: "4",
      actionType: {
        id: 1,
        name: "SMS",
      },
      actionedBy: 2,
      content:
        "Sure sir, could you provide me with your driving licence please.",
      isImportant: false,
      timestamp: new Date("06-02-2021 13:24"),
      url: "audio.mp3",
    },
    {
      id: "5",
      actionType: {
        id: 1,
        name: "carSelected",
      },
      actionedBy: 1,
      content: "Here is my driving licence",
      isImportant: false,
      timestamp: new Date("06-02-2021 13:24"),
      url: "abc.jpg",
    },
    {
      id: "6",
      actionType: {
        id: 1,
        name: "SMS",
      },
      actionedBy: 2,
      content: "Important Note",
      isImportant: true,
      timestamp: new Date("06-02-2021 13:24"),
      url: "file.pdf",
    },
  ],
  handleUnPinHistoryDetail: jest.fn(() => {}),
  handleSaveNoteDetail: jest.fn(() => {}),
  isNoteAdded: false,
};

Manager.getInstance = flexManagerMock;

jest.mock("../../services/application.service");

jest.mock("react-h5-audio-player", () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="audio-player">Audio-Player</div>;
    },
  };
});

const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
});

describe("HistoryPanel", () => {
  jest.useFakeTimers();
  it("renders correctly", async () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render timestamp for data correctly in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;

    const timeStampDiv = wrapper.findAllByProps({
      className: "time-stamp",
    });
    const timeStampItems = props.historyPanelList.map((item) => item.timestamp);

    timeStampDiv.forEach((el, index) => {
      expect(el.children).toEqual([
        Shared.getFormattedDate(timeStampItems[index], "HH:mm"),
      ]);
    });
  });

  it("should render data based on actionedBy property in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;
    const customerArea = wrapper.findAllByProps({ className: "customer-area" });
    const agentArea = wrapper.findAllByProps({ className: "agent-area" });

    expect(customerArea).toHaveLength(
      props.historyPanelList.filter(
        (item) => item.actionedBy === HistoryPanelMemberType.Customer
      ).length
    );
    expect(agentArea).toHaveLength(
      props.historyPanelList.filter(
        (item) => item.actionedBy === HistoryPanelMemberType.Agent
      ).length
    );
  });

  it("should render ImagePreview component correctly in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;
    const imagePreview = wrapper.findAllByType(ImagePreview);

    expect(imagePreview).toHaveLength(1);
  });

  it("should render AudioPlayer component correctly in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;
    const audioPlayer = wrapper.findAllByProps({ className: "audio-player" });

    expect(audioPlayer).toHaveLength(1);
  });

  it("should render file and document download component correctly in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;
    const file = wrapper.findAllByProps({ className: "document-div" });
    const downloadButton = wrapper.findAllByType(DocumentDownload);

    expect(file).toHaveLength(1);
    expect(downloadButton).toHaveLength(2);
  });

  it("should render car icon and document icon correctly in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;
    const phoneIcon = wrapper.findAllByType(PhoneInTalkIcon);
    const carIcon = wrapper.findAllByType(DirectionsCarIcon);

    expect(phoneIcon).toHaveLength(1);
    expect(carIcon).toHaveLength(1);
  });

  it("should render PinnedNote component based on isImportant property in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;

    expect(wrapper.findByType(PinnedNote)).toBeTruthy();
  });

  it("when close pin button is clicked from PinnedNote component, handleUnPinHistoryDetail method should be called", async () => {
    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    );

    expect(wrapper.find(PinnedNote).exists()).toBeTruthy();
    await act(async () => {
      const button = wrapper.find(".icon-close").at(0);
      button.simulate("click");
    });
    wrapper.update();
    expect(props.handleUnPinHistoryDetail).toHaveBeenCalled();
  });

  it("should render AddNote component correctly in UI", () => {
    const wrapper = create(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    ).root;

    expect(wrapper.findByType(AddNote)).toBeTruthy();
  });

  it("when send button is clicked from AddNote component, handleSaveNoteDetail method should be called", async () => {
    const addNoteModel: AddNoteDetailModel = {
      applicationId: 1,
      content: "new message",
      isImportant: false,
      sendToBroker: false,
    };

    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <HistoryPanel {...props} />
      </MuiThemeProvider>
    );

    expect(wrapper.find(AddNote).exists()).toBeTruthy();
    await act(async () => {
      wrapper.find(AddNote).props().onSaveNoteDetail(addNoteModel);
    });
    wrapper.update();

    expect(props.handleSaveNoteDetail).toHaveBeenCalled();
  });
});
