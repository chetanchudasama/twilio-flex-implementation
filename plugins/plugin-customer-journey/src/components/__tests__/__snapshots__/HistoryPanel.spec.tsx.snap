// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryPanel renders correctly 1`] = `
<div
  className="css-1girgli"
>
  <div
    className="heading"
  >
    HISTORY
  </div>
  <div
    className="history-container"
  >
    <div
      className="chat-container"
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <div
        className="MuiChip-root-1 date-chip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-20"
        >
          Wednesday, June, 02, 2021
        </span>
      </div>
      <div
        className="customer-area"
      >
        <div
          className="message-content-div"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-26 customer-icon-position"
          >
            description
          </span>
          Hello
          
          
          
        </div>
        <div
          className="time-stamp"
        >
          13:24
        </div>
      </div>
      <div
        className="agent-area"
      >
        <div
          className="message-content-div"
        >
          Hey, thanks for reaching out. How may I help you today?
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-35 agent-icon-position"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 15.5c-1.25 0-2.45-.2-3.57-.57-.35-.11-.74-.03-1.02.24l-2.2 2.2c-2.83-1.44-5.15-3.75-6.59-6.59l2.2-2.21c.28-.26.36-.65.25-1C8.7 6.45 8.5 5.25 8.5 4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1 0 9.39 7.61 17 17 17 .55 0 1-.45 1-1v-3.5c0-.55-.45-1-1-1zM19 12h2c0-4.97-4.03-9-9-9v2c3.87 0 7 3.13 7 7zm-4 0h2c0-2.76-2.24-5-5-5v2c1.66 0 3 1.34 3 3z"
            />
          </svg>
          
          
          
        </div>
        <div
          className="time-stamp"
        >
          13:24
        </div>
      </div>
      <div
        className="agent-area"
      >
        <div
          className="message-content-div"
        >
          Record opened, Dan Graham
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-26 agent-icon-position"
          >
            description
          </span>
          
          
          
        </div>
        <div
          className="time-stamp"
        >
          13:24
        </div>
      </div>
      <div
        className="agent-area"
      >
        <div
          className="message-content-div"
        >
          Sure sir, could you provide me with your driving licence please.
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-26 agent-icon-position"
          >
            description
          </span>
          <div
            className="agent-audio-div"
          >
            <div
              className="audio-player"
            >
              Audio-Player
            </div>
          </div>
        </div>
        <div
          className="time-stamp"
        >
          13:24
        </div>
      </div>
      <div
        className="customer-area"
      >
        <div
          className="message-content-div"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-35 customer-icon-position"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Here is my driving licence
          <div
            className="css-3r6203"
          >
            <img
              alt=""
              className="image-content"
              src="abc.jpg"
            />
          </div>
          <div
            className="css-1e48p17"
          >
            <span
              className="download-span"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-26 MuiIcon-fontSizeSmall-33 download-icon"
              >
                download
              </span>
              Download
            </span>
          </div>
        </div>
        <div
          className="time-stamp"
        >
          13:24
        </div>
      </div>
      <div
        className="agent-area"
      >
        <div
          className="message-content-div"
        >
          Important Note
          <div
            className="document-div"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-26 file-icon"
            >
              insert_drive_file
            </span>
            file.pdf
            <div
              className="css-1e48p17"
            >
              <span
                className="download-span"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-26 MuiIcon-fontSizeSmall-33 download-icon"
                >
                  download
                </span>
                Download
              </span>
            </div>
          </div>
        </div>
        <div
          className="time-stamp"
        >
          13:24
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <div
        className="css-hvu9xw"
      >
        <div
          className="pinned-chat-container"
          style={
            Object {
              "height": "100px",
            }
          }
        >
          <div
            className="MuiGrid-container-44 pinned-content"
          >
            <div
              className="MuiGrid-item-45 MuiGrid-grid-xs-1-73"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-26 MuiIcon-fontSizeSmall-33 customer-icon-position"
              >
                check_circle
              </span>
            </div>
            <div
              className="MuiGrid-item-45 MuiGrid-grid-xs-10-82 message-content"
            >
              Important Note
            </div>
            <div
              className="MuiGrid-item-45 MuiGrid-grid-xs-1-73"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-26 MuiIcon-fontSizeSmall-33 icon-close"
                onClick={[Function]}
              >
                close
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="send-message-container"
  >
    <div
      className="css-1jxatqe"
    >
      <div
        className="MuiGrid-container-44"
      >
        <div
          className="MuiGrid-item-45 MuiGrid-grid-xs-10-82 add-note-container"
        >
          <div
            className="MuiGrid-item-45 MuiGrid-grid-xs-12-84"
          >
            <div
              className="MuiFormControl-root-141 add-note-text-field"
            >
              <div
                className="MuiInputBase-root-158 MuiInput-root-145 MuiInput-underline-149 MuiInputBase-formControl-159 MuiInput-formControl-146 MuiInputBase-multiline-166 MuiInput-multiline-151"
                onClick={[Function]}
              >
                <textarea
                  aria-invalid={false}
                  className="MuiInputBase-input-168 MuiInput-input-153 MuiInputBase-inputMultiline-170 MuiInput-inputMultiline-155"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Add note"
                  required={false}
                  rows={3}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-item-45 MuiGrid-grid-xs-12-84"
          >
            <div
              className="MuiFormGroup-root-175 MuiFormGroup-row-176"
            >
              <label
                className="MuiFormControlLabel-root-177 add-note-form-control"
              >
                <span
                  className="MuiButtonBase-root-199 MuiIconButton-root-193 MuiPrivateSwitchBase-root-189 MuiCheckbox-root-183 MuiCheckbox-colorSecondary-188"
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-198"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-35 MuiSvgIcon-fontSizeSmall-42 checkbox"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-192"
                      data-indeterminate={false}
                      name="Important"
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-264"
                  />
                </span>
                <span
                  className="MuiTypography-root-202 MuiTypography-body2-210 MuiFormControlLabel-label-182"
                >
                  Important
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root-177 add-note-form-control"
              >
                <span
                  className="MuiButtonBase-root-199 MuiIconButton-root-193 MuiPrivateSwitchBase-root-189 MuiCheckbox-root-183 MuiCheckbox-colorSecondary-188"
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-198"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-35 MuiSvgIcon-fontSizeSmall-42 checkbox"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiPrivateSwitchBase-input-192"
                      data-indeterminate={false}
                      name="Share with broker"
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-264"
                  />
                </span>
                <span
                  className="MuiTypography-root-202 MuiTypography-body2-210 MuiFormControlLabel-label-182"
                >
                  Share With Broker
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-45 MuiGrid-grid-xs-2-74"
        >
          <button
            className="MuiButtonBase-root-199 MuiButtonBase-disabled-200 MuiButton-root-238 MuiButton-contained-249 MuiButton-containedPrimary-250 MuiButton-raised-252 MuiButton-raisedPrimary-253 MuiButton-disabled-258 send-btn  disable-btn"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="-1"
            type="button"
          >
            <span
              className="MuiButton-label-239"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-26 send-icon"
              >
                send
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
